networkx/algorithms/approximation/steinertree.py

```diff
@@ -217,4 +224,8 @@ def steiner_tree(G, terminal_nodes, weight="weight", method=None):
             (u, v, min(G[u][v], key=lambda k: G[u][v][k][weight])) for u, v in edges
         )
     T = G.edge_subgraph(edges)
+
+    # Remove non-terminal leaves from the generated tree
+    _remove_nonterminal_leaves(T, terminal_nodes)
+
```

Comment:
Is this necessary? Isn't `_remove_nonterminal_leaves` called as required in the functions that require it already?

Reply:
Yes it is necessary and now resolved

Reply:
I'm not convinced - isn't `_remove_nonterminal_leaves` called as needed by `algo` in L220?